plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/listeners/InlineChatFileListener.kt

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.inline.listeners
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.fileEditor.TextEditor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import software.aws.toolkits.jetbrains.services.cwc.inline.InlineChatController
+
+class InlineChatFileListener(project: Project, private val controller: InlineChatController) : FileEditorManagerListener {
```

Comment:
as will mentioned in the other PR, `Disposable` needs to be registered correctly for cleanup